hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -542,162 +543,176 @@ void prepareComputation(final Bytes input, final UnaryOperator<byte[]> aliasReso
                             precompilePricingUtils);
                     case AbiConstants.ABI_ID_REDIRECT_FOR_TOKEN -> {
                         final var target = DescriptorUtils.getRedirectTarget(input);
+                        final var isExplicitRedirectCall = target.massagedInput() != input;
+                        if (isExplicitRedirectCall) {
+                            input = target.massagedInput();
+                        }
                         final var tokenId = target.tokenId();
                         final var isFungibleToken =
-                                TokenType.FUNGIBLE_COMMON.equals(ledgers.typeOf(tokenId));
+                                (!isExplicitRedirectCall || ledgers.tokens().exists(tokenId))
```

Comment:
Why do we have this check `!isExplicitRedirectCall` for checking if its a fungibleToken ?

Reply:
Yeah, I can see how this can be confusing for a new reader. Added a comment in the code, pasting it here as well:

```
  /* For implicit redirect call scenarios, at this point in the logic it has already been
  * verified that the token exists, so comfortably call ledgers.typeOf() without worrying about INVALID_TOKEN_ID.
  * 
  *  Explicit redirect calls, however, verify the existence of the token in RedirectPrecompile.run(), so only
  *  call ledgers.typeOf() if the token exists.
  */

```